src/relay/qnn/util.h

```diff
@@ -0,0 +1,99 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*!
+ * \file src/relay/qnn/util.h
+ * \brief Utility methods needs for quantized ops that can be shared
+ */
+
+#ifndef TVM_RELAY_QNN_UTIL_H_
+#define TVM_RELAY_QNN_UTIL_H_
+
+#include <tvm/expr.h>
+#include <tvm/relay/expr.h>
+#include <limits>
+
+namespace tvm {
+namespace relay {
+namespace qnn {
+
+enum class QuantizeOpType {
+  Quantize,
+  Dequantize,
+  Requantize
+};
+
+static inline bool IsValidOpInputType(const QuantizeOpType& op_type,
```

Comment:
Maybe we can inline them into where to call it, it will be more easy to read and debug I guess.

Reply:
With recent compilers and debuggers stepping in and out of inline functions has become quite better, so I'm less fussed about it.  

To mind  the abstraction of the checker function, has the advantage that changing this on a later date means we don't have to go change all the places which are needed in case we care about higher order quantization. Actually I feel that makes the code more readable.

Reply:
Each Relay operator has a InferType function, whose responsibility is to check/set the shapes and dtypes. So, it seems like to me that there is an already a good abstraction. So, I decided to move the check to that Infer function.